lib/base/RestException.ts

```diff
@@ -9,15 +9,39 @@ export default class RestException extends Error {
 
   constructor(response) {
     super("[HTTP " + response.statusCode + "] Failed to execute request");
+    const [body, bodyParseError] = parseResponseBody(response.body);
 
-    const body =
-      typeof response.body === "string"
-        ? JSON.parse(response.body)
-        : response.body;
     this.status = response.statusCode;
-    this.message = body.message;
-    this.code = body.code;
-    this.moreInfo = body.more_info; /* jshint ignore:line */
-    this.details = body.details;
+    if (body) {
+      this.message = body.message;
+      this.code = body.code;
+      this.moreInfo = body.more_info; /* jshint ignore:line */
+      this.details = body.details;
+    }
+    if (body === null) {
+      this.message = "Failed to parse response body";
+      this.code = 12100;
+      this.moreInfo = "https://www.twilio.com/docs/api/errors/12100";
```

Comment:
This does fit right since it's not a 12100 or even necessarily an error produced by Twilio, like in the case of a [451](https://github.com/twilio/twilio-node/issues/570#issuecomment-623019985).

Reply:
i wont need the body null check if the population is confirmed. Will fix.

Reply:
what is the correct code when its a bad json response?
do i still use   "https://www.twilio.com/docs/errors/81022"?

Reply:
I don't think either field should be populated if the response is not an object or a JSON string. It likely didn't come from Twilio.

Reply:
@childish-sambino 
based on the commend:
* updated the assumptions written in the PR.
* changed unit test to expect the instanced RestException to contain undefined member variables when the received response_body is a malformed json string 
* simplified the helper function
* introduce a contract to couple the RestException Class to the private helper function. (implements ResponseError)
* Because the exception does not get overwritten when a malformed json response_body string is recieved, the  exception.message retains the supered message: `super("[HTTP " + response.statusCode + "] Failed to execute request")`